src/components/Posts.tsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect } from 'react';
+import * as postsActions from '../features/posts';
+import { PostsList } from './PostsList';
+import { Loader } from './Loader';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { User } from '../types/User';
+
+export const Posts: React.FC = () => {
+  const dispatch = useAppDispatch();
+
+  const { posts, hasError, loaded } = useAppSelector(state => state.posts);
+  const author = useAppSelector<User | null>(state => state.author.author);
+
+  useEffect(() => {
+    dispatch(postsActions.setSelectedPost(null));
+
+    if (author?.id) {
+      dispatch(postsActions.initPosts(author.id));
+    }
+  }, [author?.id]);
+
+  return (
+    <div className="block" data-cy="MainContent">
+      {!author && (
+        <p data-cy="NoSelectedUser">
+          No user selected
+        </p>
+      )}
+
+      {author && !loaded && (
+        <Loader />
+      )}
+
+      {author && loaded && hasError && (
+        <div
+          className="notification is-danger"
+          data-cy="PostsLoadingError"
+        >
+          Something went wrong!
+        </div>
+      )}
+
+      {author && loaded && !hasError && posts.length === 0 && (
```

Comment:
```suggestion
      {author && loaded && !hasError && !posts.length && (
```
pls consider moving long ternary expressions to separate consts everywhere in your project